File: LeetCodeCPlusPlus/py58.cpp
// 最后一个单词的长度
#include <string>
using namespace std;
class Solution {
public:
    int lengthOfLastWord(string s) {
        int end = s.size() - 1;
        while (end>=0 && s[end] == ' ')
            end--;
        int start = end;
        while (start>=0 && s[start] != ' ')
            start--;
        return end-start; 
    }
};

File: LeetCodePython/py58.py
# 最后一个单词的长度
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        if not s:
            return 0
        i = len(s) - 1
        while i>=0 and s[i] == ' ':
            i -= 1
        if i == -1:
            return 0
        ans = ''
        while i>=0 and s[i] != ' ':
            ans = s[i] + ans
            i -= 1
        return len(ans)